scripts/peer/peer.py

```diff
@@ -0,0 +1,273 @@
+"""
+Test run this with:
+
+    cd <dir where peer.py file is>
+    analysis-runner \
+        --access-level test \
+        --description 'Test run peer analysis' \
+        --output-dir 'peer/2022-03-21' \
+        --dataset tob-wgs \
+        python3 peer.py \
+        --path-to-cell-files 'gs://cpg-tob-wgs-test/scrna-seq/grch38_association_files/expression_files/'
+"""
+
+
+import os
+import logging
+import click
+import hailtop.batch as hb
+import pandas as pd
+from cpg_utils.hail import remote_tmpdir, output_path
+from google.cloud import storage
+
+PEER_DOCKER = 'australia-southeast1-docker.pkg.dev/cpg-common/images/peer:1.3.2'
+SCORES_PATH = 'gs://cpg-tob-wgs-test/kat/pca/nfe_feb22/v0/scores.json'
+COVARIATES_PATH = 'gs://cpg-tob-wgs-test/scrna-seq/grch38_association_files/covariates_files/covariates.tsv'
+SAMPLE_ID_KEYS_PATH = 'gs://cpg-tob-wgs-test/scrna-seq/grch38_association_files/metadata/keys_metadata_sheet.csv'
+
+
+def get_covariates(
+    scores_path, covariates_path, expression_file, sample_id_keys_path
+) -> str:
+    """
+    Get covariate data by merging PCA scores with age and sex info.
+    Only needs to be run once. This returns a TSV (as a string)
+    """
+    # load in scores which have outliers removed
+    scores_df = pd.read_json(scores_path)
+    sampleid = scores_df.s
+    # only get the first 4 PCs, as indicated by scree plot
+    scores = pd.DataFrame(scores_df['scores'].to_list()).iloc[:, 0:4]
+    scores.columns = ['PC1', 'PC2', 'PC3', 'PC4']
+    scores.insert(loc=0, column='sampleid', value=sampleid)
+    # filter to only CPG samples
+    scores = scores[scores.sampleid.str.contains('CPG')]
+    print(len(scores.index))
+    # 1006
+    # change CPG IDs to One1K1K IDs
+    sampleid_keys = pd.read_csv(sample_id_keys_path, sep=',')
+    scores = pd.merge(
+        scores, sampleid_keys, how='left', left_on='sampleid', right_on='CPG_ID'
+    )
+    print(len(scores.OneK1K_ID.dropna()))
```

Comment:
ditto, I'll stop highlighting them but there are a couple more.

Reply:
I've used the logging library as well